sim/midas/src/main/scala/midas/passes/fame/FAMETransform.scala

```diff
@@ -147,35 +147,82 @@ object FAMEModuleTransformer {
      */
     val nReset = DoPrim(PrimOps.Not, Seq(WRef(hostReset)), Seq.empty, BoolType)
 
+    /**
+      * Bundles together four critical elements for managing multiclock processing.
+      *
+      * @param Port The reference to the original target port
+      *
+      * @param outputChannelEnable A boolean that, when asserted alongside
+      * targetCycleFinishing, serves to reset output channels in the associated
+      * clock domain so that new output tokens may be enqueued.
+      *
+      * @param inputChannelEnable Ditto, above. This is asserted one pipeline
+      * stage earlier, ensuring that all inputs are dequeued synchronously with
+      * the launching of the target clock.
```

Comment:
I am not really sure what the comments for `outputChannelEnable` / `inputChannelEnable` are saying.

As far as what the enables do, I think that "reset [input|output] channels" is maybe unclear vs. "lets an [input|output] channel synchronous to this clock [consume|produce] a token in this time step," since the reset as an FSM transition is just a convenient FIRRTL code golf thing.

And maybe it might make more sense to reverse the description and say that the `outputChannelEnable` is a pipelined version of `inputChannelEnable`.